komposable-architecture/src/test/java/com/toggl/komposable/reducer/forEach/ForEachTestUtils.kt

```diff
@@ -0,0 +1,30 @@
+package com.toggl.komposable.reducer.forEach
+
+import com.toggl.komposable.architecture.ReduceResult
+import com.toggl.komposable.architecture.Reducer
+import com.toggl.komposable.extensions.noEffect
+
+sealed class ParentAction {
+    data class EditText(val text: String) : ParentAction()
+    data class ElementActionWrapper(val id: Int, val elementAction: ElementAction) : ParentAction()
+}
+
+sealed class ElementAction {
+    data class EditText(val text: String) : ElementAction()
+}
+
+data class ElementState(val elementText: String)
```

Comment:
Calling this class `ElementState` makes it really hard to wrap one's head around what should happen. Using a less generic name (like just `Element` or maybe even `Item`) could improve readability for the tests.

Even better would be to change these into a concrete example, so real world usage would be easier to grasp for someone trying to understand this concept from the tests. TimeEntries feel like a very natural example. Another good one would be a list of songs/albums

Reply:
`ElementState` is derived from the generic name used in forEach. But yeah, tests might not be the best way to wrap one's head around this concept; that's where sample apps and case studies come in handy. I've started working on the new TODO app, where this should be demonstrated quite effectively.